base_layer/core/src/transaction.rs

```diff
@@ -208,6 +213,28 @@ impl TransactionKernel {
         self.lock_height = lock_height;
         self
     }
+
+    pub fn verify_signature(&self) -> Result<(), TransactionError> {
+        if self.excess.is_some() && self.excess_sig.is_some() {
+            let signature = self.excess_sig.unwrap();
+            let excess = self.excess.unwrap();
+            let excess = excess.as_public_key();
+            let r = signature.get_public_nonce();
+            let c = Challenge::<Blake256>::new()
```

Comment:
Should be not rather replace this with a generic digest?

Reply:
Also this uses Blake256, are we not doing 2b?

Reply:
It is what the ristretto_signature.rs implementation is using.

Reply:
Actually I think for the challenge this is chosen specifically because the challenge needs to be 256 bits. Blake256 is the specific version of Blake that produces that length output.

Reply:
I dont really care about the specific hash implementation we use, 
I just think we should abstract that digest trait up to implementation level and not hard code it here.

Reply:
So it is dictated by the underlying signature implementation in the crypto crate, so maybe the generic should not actually be exposed at all. I don't see the value in making it generic at the transaction level? Then every time you want to make a transaction you need to specify the hashing function used in the signature implementation.

Reply:
So I added a type alias for the signature hash to the type.rs file so that you can set choose this hash function along with choosing the other concrete implementations for this crate.